hydra-node/test/Hydra/API/ServerSpec.hs

```diff
@@ -267,9 +273,72 @@ spec = describe "ServerSpec" $
                     Right (timedOutputs :: [TimedServerOutput SimpleTx]) ->
                       seq <$> timedOutputs `shouldSatisfy` strictlyMonotonic
 
+    it "displays correctly headStatus and snapshotUtxo in a Greeting message" $
+      showLogsOnFailure $ \tracer ->
+        withFreePort $ \port -> do
+          withAPIServer @SimpleTx "127.0.0.1" port alice mockPersistence tracer noop $ \Server{sendOutput} -> do
+            let status v = v ^? key "headStatus"
+            let snapshotUtxos v = v ^? key "snapshotUtxo"
+            let generateSnapshot =
+                  generate $
+                    SnapshotConfirmed <$> arbitrary <*> arbitrary <*> arbitrary
+            waitForValue port $ \v -> do
+              guard $ status v == Just (Aeson.String "Idle")
+              -- test that the 'snapshotUtxo' is excluded from json if there is no utxo
+              guard $ isNothing (v ^? key "snapshotUtxo")
```

Comment:
Could: improve test assertions / errors by using `shouldBe` where possible.

I think it would be better to wait for any `Greetings` message and then do the assertions explicitly. Sorry I did not suggest this earlier.

```suggestion
            greeting1 <- waitForValue port $ \v ->
              if matchGreeting v
              then pure v
              else Nothing
            greeting1 ^. key "headStatus `shouldBe` Just (Aeson.String "Idle")
            -- test that the 'snapshotUtxo' is excluded from json if there is no utxo
            greeting1 ^. key "snapshotUtxo" `shouldBe` Nothing
```

There is likely a better way to do the match in `waitForValue` or to reduce this boilerplate.
However, this will be more explicit if one or the other field is not of the expected value.